transient/transient.py

```diff
@@ -379,20 +372,15 @@ def run(self) -> None:
                 "ssh_config": self.ssh_config,
                 "local_dir": absolute_local_path,
                 "remote_dir": remote,
-                "local_user": self.__current_user(),
             }
 
-            # Slamming the server with 20 connections at once is a good way to break things:
-            if len(sshfs_threads) > MAX_CONCURRENT_SSHFS:
-                sshfs_threads.pop(0).join()
-
-            sshfs_threads.append(
-                checked_threading.Thread(target=sshfs.do_sshfs_mount, kwargs=sshfs_kwargs)
-            )
+            sshfs_threads.append(sshfs.SshfsThread(**sshfs_kwargs))
             sshfs_threads[-1].start()
 
         for sshfs_thread in sshfs_threads:
-            sshfs_thread.join()
+            sshfs_thread.is_complete.wait()
```

Comment:
I'd rather have a wait method on the thread that waits on is_complete and does the lines below